app/src/main/java/com/eveningoutpost/dexdrip/StartNewSensor.java

```diff
@@ -83,42 +89,24 @@ public String getMenuName() {
     }
 
     public void addListenerOnButton() {
-
         button = (Button)findViewById(R.id.startNewSensor);
         linkPickers = (CheckBox)findViewById(R.id.startSensorLinkPickers);
         SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
         linkPickers.setChecked(prefs.getBoolean("start_sensor_link_pickers", false));
 
         button.setOnClickListener(new View.OnClickListener() {
-          public void onClick(View v) {
-
-              Calendar calendar = Calendar.getInstance();
-              calendar.set(dp.getYear(), dp.getMonth(), dp.getDayOfMonth(),
-              tp.getCurrentHour(), tp.getCurrentMinute(), 0);
-              long startTime = calendar.getTime().getTime();
-
-              Sensor.create(startTime);
-              Log.d("NEW SENSOR", "Sensor started at " + startTime);
-
-              Toast.makeText(getApplicationContext(), "NEW SENSOR STARTED", Toast.LENGTH_LONG).show();
-              
-              SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
-              prefs.edit().putBoolean("start_sensor_link_pickers", linkPickers.isChecked()).apply();
-              
-              CollectionServiceStarter.newStart(getApplicationContext());
-              Intent intent;
-              if(prefs.getBoolean("store_sensor_location",true)) {
-                  intent = new Intent(getApplicationContext(), NewSensorLocation.class);
-              } else {
-                  intent = new Intent(getApplicationContext(), Home.class);
-              }
-
-              startActivity(intent);
-              finish();
-          }
-
+            public void onClick(View v) {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
I think this should only be done for G5.
It is missing for other InputMethods as well but they have a different logic for connection (-> permission request in bt scan activity).